lib/services/static-files-api-service.js

```diff
@@ -0,0 +1,155 @@
+                    });
+                })
+                .catch(function(e) {
+                    logger.debug("Directory " + skuPath + "/" + item + " is not in the database",
+                        e);
+                    return result;
+                });
+            },{}).then(function(pairs) {
+                dirs.push(skuPath);
+                return pairs;
+            });
+        })
+        .catch(function(e) {
+            logger.debug("StaticFilesApiService.pairSkupackIds", e);
+            return {};
+        });
+    };
+
+
+    StaticFilesApiService.prototype.walkDirectory = function(dir) {
+        return Promise.try(function() {
+            return new Promise(function(resolve, reject) {
+                var walkResults = [];
+                walker = walk.walk(dir, {followLinks: false});
+
+                walker.on("file", function(root, fileStats, next) {
+                    Promise.resolve().then(function() {
+                        var walkObj = {};
+                        if (root.indexOf(userPath) > -1){
+                            root = root.replace(userPath, '');
+                            walkObj.uri = path.join(root, fileStats.name);
+                            walkResults.push(walkObj);
+                        } else if (root.indexOf(staticPath) > -1){
+                            var longPath = path.join(root, fileStats.name).split("/");
+                            var httpIndex = longPath.indexOf("http") + 1;
+                            walkObj.uri = longPath.slice(httpIndex).join("/");
+                            walkResults.push(walkObj);
+                        } else if ((root.indexOf(skuPath) > -1) && 
+                            (root.indexOf("static") > -1)){
+                            var filePath = path.join(root, fileStats.name).split("/");
+                            var skuId = filePath[1];
+                            var skuUri = filePath.slice(3).join("/");
+                            walkObj.uri = skuUri;
+                            walkObj.sku = skuId;
+                            walkResults.push(walkObj);
+                        }
+                    }).then(function() {
+                        next();
+                    }).catch(function(e) {
+                        reject(e);
+                    });
+                });
+
+                walker.on("end", function() {
+                    resolve(walkResults);
+                });
+            });
+        })
+        .catch(function(e) {
+            logger.debug("walkDirectory was provided with a bad path: " + dir, e);
+            return [];
+        });
+    };
+
+
+    StaticFilesApiService.prototype.getAllStaticFiles = function() {
+
+        var self = this;
+        var directories = [staticPath,
+                           configuration.get('httpStaticRoot', userPath)];
+
+        return fs.lstatAsync(skuPath).then(function(stat) {
+            stat.isDirectory();
```

Comment:
Good catch, those lines have been removed and the inclusion of the skuPath directory has been fixed.